src/gtest/test_joinsplit.cpp

```diff
@@ -22,6 +22,39 @@ using namespace libzcash;
 
 extern ZCJoinSplit* params;
 
+typedef std::vector<JSDescription> SproutProofs;
+// Make both the PHGR and Groth proof for a Sprout statement,
+// and store the results in JSDescription objects.
+SproutProofs makeSproutProofs(
+        const std::array<JSInput, 2>& inputs,
+        const std::array<JSOutput, 2>& outputs,
+        const uint256& joinSplitPubKey,
+        uint256& out_randomSeed,
+        uint64_t vpub_old,
+        uint64_t vpub_new,
+        const uint256& rt
+){
+    //Making the PHGR proof
+    JSDescription phgr(false, *params, joinSplitPubKey, rt, inputs, outputs, vpub_old, vpub_new);
+    //Making the Groth proof
+    JSDescription groth(true, *params, joinSplitPubKey, rt, inputs, outputs, vpub_old, vpub_new);
```

Comment:
This is called from `test_full_api()` which takes `ZCJoinSplit* js` as a parameter rather than using the global. Pass that `js` parameter through and use it here (and likewise in `verifySproutProofs()`).

Reply:
Use `js` here (like `verifySproutProofs()` is now doing).